db/db.go

```diff
@@ -71,12 +73,20 @@ type db struct {
 
 	// The ID of the last transaction created.
 	previousTxnID atomic.Uint64
+
+	// ACP module
+	acp immutable.Option[acp.ACPModule]
 }
 
 // Functional option type.
 type Option func(*db)
 
-const updateEventBufferSize = 100
+// WithACPModule enables access control. If path is empty, then acp module runs in-memory.
+func WithACPModule(path string) Option {
```

Comment:
question: The initial ACP PR will only introduce a local variant of ACP, it is not making remote calls?

Reply:
Correct.